Tests/SwiftDocCTests/Infrastructure/PathHierarchyTests.swift

```diff
@@ -3126,7 +3126,103 @@ class PathHierarchyTests: XCTestCase {
         try assertFindsPath("/MainModule/TopLevelProtocol/extensionMember(_:)", in: tree, asSymbolID: "extensionMember1")
         try assertFindsPath("/MainModule/TopLevelProtocol/InnerStruct/extensionMember(_:)", in: tree, asSymbolID: "extensionMember2")
     }
-    
+
+    func testAbsoluteLinksToOtherModuleWithExtensions() async throws {
+        enableFeatureFlag(\.isExperimentalLinkHierarchySerializationEnabled)
+
+        let importedProtocolID = "s:14ImportedModule12BaseProtocolP"
+        let importedTypeID = "s:14ImportedModule12ExtendedTypeV"
+        let extensionSymbolID = "s:e:s:14ImportedModule12ExtendedTypeV04MainC0E15extensionMethodyyF"
+        let extensionMethodID = "s:14ImportedModule12ExtendedTypeV04MainC0E15extensionMethodyyF"
+        let mainModuleTypeID = "s:10MainModule0A4TypeV"
+
+        let extensionMixin = SymbolGraph.Symbol.Swift.Extension(
+            extendedModule: "ImportedModule",
+            typeKind: .struct,
+            constraints: []
+        )
+
+        let catalog = Folder(name: "TestCatalog.docc", content: [
+            JSONFile(name: "MainModule.symbols.json", content: makeSymbolGraph(
+                moduleName: "MainModule",
+                symbols: [
+                    makeSymbol(id: mainModuleTypeID, kind: .struct, pathComponents: ["MainType"])
+                ]
+            )),
+            JSONFile(name: "MainModule@ImportedModule.symbols.json", content: makeSymbolGraph(
+                moduleName: "MainModule",
+                symbols: [
+                    makeSymbol(id: importedProtocolID, kind: .protocol, pathComponents: ["BaseProtocol"]),
+                    makeSymbol(id: importedTypeID, kind: .struct, pathComponents: ["ExtendedType"]),
```

Comment:
AFAIK extension symbol graph files (ExtendingName@ExtendedName) don't include the definition of the symbols being extended. Those symbol definitions would belong in the main symbol graph for the ExtendedName module instead.

```suggestion
```